sonic-net/sonic-buildimage
# [Marvell] Build infrastructure enhancements

#### Why I did it
This change is about marvell build infrastructure enhancement by merging architecture specific marvell platform(marvell-arm64 and marvell-armhf) folders under 'marvell'.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
This is archived with following changes divided into following commits,
* Move mrvl-prestera and sonic-platform-marvell submodule 
   'mrvl-prestera' and 'sonic-platform-marvell' submodules are moved from 'marvell-arm64'to 'marvell'. Additionaly submodules are changed to support multi-architecture build. 

* Create sonic-platform-nokia from marvell-arm64/armhf
   'sonic-platform-nokia' is created from merge of 'marvell-armhf' and 'marvell-arm64' in 'marvell'. Additionally changes are done in debian/rules to support multi-architecture build.

* Change marvell platform makefile to support all arch
  Combine all the architecture targets in makefiles. Added CONFIGURED_ARCH check to build for specific architecture.
  Updated sai deb which matches latest SAI commit.

* Add support for architecture specific platform.conf
  Changed build_image.sh and onie-mk-demo.sh to read architecture specific platform.conf.
  Moved marvell-arm64/platform.conf to marvell/platform_arm64.conf.
  Moved marvell-armhf/platform.conf to marvell/platform_armhf.conf.

* [arm64] Adjust the path to read sonic_fit.its
   Use CONFIGURED_PLATFORM based path while reading sonic_fit.its.

* [arm64/armhf] Add architecure suffix for output image
  This change is needed to generate the output image with architecture suffix for arm64/armhf platform.
  For arm64, defined a marvell specific file to avoid affect on other vendors.

* Change platform_asic name to marvell
  Change 'marvell-arm64' and 'marvell-armhf' platform_asic name to 'marvell'.

* Remove marvell-armhf and marvell-arm64 folders
  Removed 'marvell-armhf' and 'marvell-arm64' as infra is combined with 'marvell' folder.

* Update readme to reflect new build commands

* Change azure pipeline commands for marvell

New build commands:
AMD64 - make configure PLATFORM=marvell && make  target/sonic-marvell.bin
ARM64 - make configure PLATFORM=marvell PLATFORM_ARCH=arm64 && make target/sonic-marvell-arm64.bin
ARMHF - make configure PLATFORM=marvell PLATFORM_ARCH=armhf && make  target/sonic-marvell-armhf.bin

#### How to verify it
Built images for all the platform and installed on hardware to verify the port up.
Verified onie and sonic-sonic install.



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
